flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/PartitionMonitorTest.java

```diff
@@ -0,0 +1,203 @@
+                        StorageDescriptor sd = partition.getSd();
+                        Map<String, Object> partitionColValues = new HashMap<>();
+                        for (String partCol : partition.getValues()) {
+                            String[] arr = partCol.split("=");
+                            Asserts.check(
+                                    arr.length == 2, "partition string should be key=value format");
+                            partitionColValues.put(arr[0], arr[1]);
+                        }
+                        return new HiveTablePartition(sd, partitionColValues, new Properties());
+                    }
+
+                    @Override
+                    public TypeSerializer<Long> getTypeSerializer() {
+                        return super.getTypeSerializer();
+                    }
+
+                    @Override
+                    public Long getConsumeStartOffset() {
+                        return super.getConsumeStartOffset();
+                    }
+
+                    @Override
+                    public void close() throws Exception {
+                        super.close();
+                    }
+
+                    @Override
+                    public void open() throws Exception {
+                        super.open();
+                    }
+
+                    @Override
+                    public List<ComparablePartitionValue> getComparablePartitionValueList()
+                            throws Exception {
+                        return super.getComparablePartitionValueList();
+                    }
+                };
+
+        ContinuousPartitionFetcher<Partition, Long> continuousPartitionFetcher =
+                new ContinuousPartitionFetcher<Partition, Long>() {
+
+                    @Override
+                    public List<Tuple2<Partition, Long>> fetchPartitions(
+                            Context<Partition, Long> context, Long previousOffset)
+                            throws Exception {
+                        return testPartitionWithOffset.stream()
+                                .filter(p -> Long.valueOf(p.getCreateTime()) >= previousOffset)
+                                .map(p -> Tuple2.of(p, Long.valueOf(p.getCreateTime())))
+                                .collect(Collectors.toList());
+                    }
+
+                    @Override
+                    public List<Partition> fetch(PartitionFetcher.Context<Partition> context)
+                            throws Exception {
+                        return null;
+                    }
+                };
+
+        partitionMonitor =
+                new ContinuousHiveSplitEnumerator.PartitionMonitor(
+                        currentReadOffset,
+                        seenPartitionsSinceOffset,
+                        tablePath,
+                        jobConf,
+                        continuousPartitionFetcher,
+                        fetcherContext);
```

Comment:
The requirement of PartitionMonitor is just a context with `toHiveTablePartition`, I think you can create `HiveContext` interface, or split `toHiveTablePartition` to another interface.